controllers/tmf-apis/usageManagement.js

```diff
@@ -0,0 +1,138 @@
+var AccountingService = require('./../../db/schemas/accountingService'),
+    async = require('async'),
+    storeClient = require('./../../lib/store').storeClient,
+    utils = require('./../../lib/utils');
+
+var usageManagement = ( function () {
+
+    /**
+     * Check if the API Key passed is a valid API Key
+     *
+     * @param  {Object}   req      Incoming request.
+     */
+    var validateApiKey = function (req, callback) {
+
+        var apiKey = req.get('X-API-KEY');
+
+        if (!apiKey) {
+
+            return callback({
+                status: 401,
+                message: 'Missing header "X-API-KEY"'
+            });
+
+        } else {
+
+            AccountingService.findOne( {apiKey: apiKey}, function (err, result) {
+
+                if (err) {
+
+                    return callback({
+                        status: 500,
+                        message: 'Error validating apiKey'
+                    });
+
+                } else if (!result) {
+
+                    return callback({
+                        status: 401,
+                        message: 'Invalid apikey'
+                    });
+
+                } else if (result.state !== 'COMMITTED') {
+
+                    return callback({
+                        status: 401,
+                        message: 'Apikey uncommitted'
+                    });
+
+                } else {
+                    return callback();
+                }
+            });
+        }
+    };
+
+    /**
+     * Assigns the user id to the relatedPaty query string if it is not defined.
+     *
+     * @param  {Object}   req      Incoming request
+     */
+    var validateRetrieving = function (req, callback) {
+
+        var userId = req.user.id;
+
+        if (!req.query.relatedParty) {
+
+            req.query.relatedParty = {
```

Comment:
This won't work. You are trying to include a query parameter in the request that is done to the final API but it won't be included because the implemented logic in `tmf.js` does not work in this way. You have two options:

* Modify `tmf.js` so the final query string is formed based on `req.query`. This will imply modifying some controllers (recommended).
* Modify `apiUrl` manually to include the new query param. This is the way we have been using until now. You can take advantage of the function `filterRelatedPartyFields` in `lib/tmfUtils.js` that does this work for you! :)